### PR TITLE
Feat/authentication authorization

### PR DESCRIPTION
**Pull request by shamir92**
## Summary by CodeRabbit

- **New Features**
  - Added `UserRole` type with constants for "IT" and "NURSE" in the `User` struct.
  - Updated `ValidateNIP` to validate based on user role.

- **Bug Fixes**
  - `GetBcryptSalt` now returns a default value if the salt is not a number instead of logging a fatal error.

- **Refactor**
  - Renamed `InsertUserIT` method to `InsertUser` and updated its usage.
  - Adjusted `RegisterUserIT` to include role validation and setting.
  - Moved instantiation of `bcrypt` and `jwtManager` below the error handling block in `main()`.

- **Chores**
  - Updated `.gitignore` to exclude `w3/bin/air`.
  - Removed unnecessary logging in `postgres_writer.go`.

### CODE REVIEW

**Review by 14mdzk — APPROVED** [review]

File: w3/domain/usecase/user_it_usecase.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func (u *userITUsecase) RegisterUserIT(userITRequest UserITRegisterRequest) (Use
 	var user entity.User
 
 	userNip := strconv.FormatInt(int64(userITRequest.NIP), 10)
-	if !user.ValidateNIP(userNip) {
+	if !user.ValidateNIP(userNip, "IT") {
```
Comment (enjaytarigan) [inline w3/domain/usecase/user_it_usecase.go:45]: use enum from user_entity.go

File: w3/domain/usecase/user_it_usecase.go [hunk 2]
```diff
@@ -58,8 +58,9 @@ func (u *userITUsecase) RegisterUserIT(userITRequest UserITRegisterRequest) (Use
 	user.Name = userITRequest.Name
 	user.NIP = userNip
 	user.Password = hashedPassword
+	user.Role = "IT"
```
Comment (enjaytarigan) [inline w3/domain/usecase/user_it_usecase.go:61]: use enum from user_entity.go

**Review by enjaytarigan — CHANGES_REQUESTED** [review]

File: w3/domain/usecase/user_it_usecase.go (hunk 1 above)
Comment (shamir92) [reply w3/domain/usecase/user_it_usecase.go]: DONE

File: w3/domain/usecase/user_it_usecase.go (hunk 2 above)
Comment (shamir92) [reply w3/domain/usecase/user_it_usecase.go]: DONE

**Review by enjaytarigan — APPROVED** [review]
LGTM

**Pull request merged**